lambdas/question/src/test/java/uk.gov.di.ipv.cri.kbv.api.handler/QuestionHandlerTest.java

```diff
@@ -805,6 +806,70 @@ void shouldNotSendThinFileEncounteredAuditEventGivenResponsesStatusIsNull()
+                    ArgumentCaptor.forClass(Map.class);
+            ArgumentCaptor<AuditEventContext> argumentCaptorForResponseReceived =
+                    ArgumentCaptor.forClass(AuditEventContext.class);
+            ArgumentCaptor<Map<String, Object>> auditEventMapForResponseReceived =
+                    ArgumentCaptor.forClass(Map.class);
+
+            QuestionState questionState = new QuestionState();
+            KBVItem kbvItem = new KBVItem();
+            UUID sessionId = UUID.randomUUID();
+            SessionItem sessionItem = new SessionItem();
+            sessionItem.setSessionId(sessionId);
+            kbvItem.setSessionId(sessionId);
+            PersonIdentityDetailed personIdentity = mock(PersonIdentityDetailed.class);
+
+            when(mockPersonIdentityService.getPersonIdentityDetailed(sessionId))
+                    .thenReturn(personIdentity);
+            QuestionsResponse experianQuestionResponse = getAttemptAlertResponse();
+            doReturn(experianQuestionResponse)
+                    .when(spyKBVService)
+                    .getQuestions(any(QuestionRequest.class));
+
+            when(mockConfigurationService.getParameterValue(IIQ_STRATEGY_PARAM_NAME))
+                    .thenReturn(MOCK_IIQ_STRATEGY_PARAM_VALUE);
+            when(mockObjectMapper.readValue(
+                            anyString(), Mockito.<TypeReference<Map<String, String>>>any()))
+                    .thenReturn(MOCK_IIQ_STRATEGY_MAPPED_VALUE);
+            when(mockConfigurationService.getVerifiableCredentialIssuer())
+                    .thenReturn("kbv-component-id");
+
+            assertNotNull(
+                    questionHandler.processQuestionRequest(
+                            questionState, kbvItem, sessionItem, requestHeaders));
+
+            verify(mockPersonIdentityService).getPersonIdentityDetailed(kbvItem.getSessionId());
+            verify(mockAuditService)
+                    .sendAuditEvent(
+                            eq(THIN_FILE_ENCOUNTERED.toString()),
+                            argumentCaptorForThinFile.capture(),
+                            auditEventMapForThinFile.capture());
+            verify(mockAuditService)
+                    .sendAuditEvent(
+                            eq(AuditEventType.RESPONSE_RECEIVED),
+                            argumentCaptorForResponseReceived.capture(),
+                            auditEventMapForResponseReceived.capture());
+            assertEquals(sessionItem, argumentCaptorForThinFile.getValue().getSessionItem());
+            assertEquals(requestHeaders, argumentCaptorForThinFile.getValue().getRequestHeaders());
+            Map<?, ?> auditEventExtensionEntries =
+                    (Map<?, ?>)
+                            auditEventMapForResponseReceived.getValue().get("experianIiqResponse");
+            assertEquals(
+                    Map.of("outcome", "Unable to Authenticate", "repeatAttemptAlert", true),
```

Comment:
Is there a way to distinguish a true thinfile from a sudo thinfile (i.e. one that's not actually a thinfile but as a result of repeat attempts) Perhaps the logic in thinfile should be exclusive to include condition where repeat alert is false. If repeatAlert is true then it's a true repeat alert. I think this should not be classed as thinFile

Thinfile should be - "Unable to Authenticate AND repeatAttemptAlert = false"
Repeat alert should be - "Unable to Authenticate AND repeatAttemptAlert = true"

Reply:
Yep this test is just specifically for a thin file with an Alert. I'll refactor/rename this test though as it is confusing.

Reply:
So from what I can see (may be more logic I have not seen). Any `Unable to Authenticate` AuthenticationResult in an SAA response will produce a thin file audit event: [isThinFile logic](https://github.com/govuk-one-login/ipv-cri-kbv-api/blob/main/lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/domain/QuestionsResponse.java#L72)

The only change included in this PR is checking if `<Alerts>` is also in the SAA response, it will add that to the RESPONSE_RECEIVED audit event if so.

Reply:
^ This has changed as we now the Alert comes in the SAAResponse with the questions. Repeat alerts can come with Questions and Thinfile responses (tested in UAT).